apps/vr-tests-react-components/src/utilities/getStoryVariant.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import type { Args, StoryFn, StoryObj } from '@storybook/react';
+import { FluentProvider } from '@fluentui/react-provider';
+import { webLightTheme, webDarkTheme, teamsHighContrastTheme } from '@fluentui/react-theme';
+
+export const DARK_MODE = 'Dark Mode';
+export const HIGH_CONTRAST = 'High Contrast';
+export const RTL = 'RTL';
+
+type StoryVariant = typeof DARK_MODE | typeof HIGH_CONTRAST | typeof RTL;
+
+/** Helper function that returns RTL, Dark Mode or High Contrast variant of an existing story. */
+export function getStoryVariant<TArgs = Args>(story: StoryFn<TArgs>, variant: StoryVariant): StoryObj<TArgs> {
```

Comment:
this is incorrectly set generic as `TArgs` will never be properly inferred based on usage.

also the need to use `!` assertion within the test points on a type interface issue

we have 2 options how to mitigate this:

**Option 1:**

keep explicit return type extended by manually build return type from custom parameters

```
export function getStoryVariant<TArgs extends Args>(
  story: StoryFn<TArgs>,
  variant: StoryVariant,
): StoryObj<TArgs> & {
  parameters: { dir: ReturnType<typeof getDir>; mode: 'vr-test'; theme: ReturnType<typeof getTheme> };
}
```

**Option 2:**

use TS inference to properly infer the return value of the function

```ts
export function getStoryVariant(story: StoryFn, variant: StoryVariant) {
 
// omitted code

  return { /* ...  omitted code */ } satisfies StoryObj;
}
```

---

both solution can completely remove the generic because issue I mentioned initially - we can keep it if we have any usage where we need to manually instatiate the `TArgs` type in order to get strictly typed Story Object

Reply:
I prefer Option 2; I've just committed it. Thanks for the suggestion!